app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -101,6 +110,7 @@ def alternate_machine_id(machine_id)
         # providers store it in downcase, so, we let the upcase/downcase logic to other methods with more
         # business knowledge.
         # 1 - https://bugzilla.redhat.com/show_bug.cgi?id=1294461
+        return if machine_id.length < 31
```

Comment:
If the method `find_host_by_bios_uuid` received `nil` as a parameter it returns `nil` immediately, it doesn't try to find anything, so if the machine_id is not valid (line 43), host_instance will be nil.